### PR TITLE
BigQuery: option for job ID generation with user-supplied prefix

### PR DESCRIPTION
**Pull request by alixhami**

### CODE REVIEW

File: bigquery/google/cloud/bigquery/client.py [hunk 1]
```diff
@@ -611,6 +611,11 @@ def load_table_from_storage(self, source_uris, destination,
         :type job_id: str
         :param job_id: (Optional) Name of the job.
 
+        :type job_id_prefix: str or ``NoneType``
```
Comment (jba) [inline bigquery/google/cloud/bigquery/client.py]: remove the first "job ID:"

"the user-provided prefix for a ..."

File: bigquery/google/cloud/bigquery/client.py [hunk 2]
```diff
@@ -670,7 +680,7 @@ def load_table_from_file(self, file_obj, destination,
                  be determined, or if the ``file_obj`` can be detected to be
                  a file opened in text mode.
         """
-        job_id = _make_job_id(job_id)
+        job_id = _make_job_id(job_id, prefix=job_id_prefix)
```
Comment (jba) [inline bigquery/google/cloud/bigquery/client.py]: (nit) I don't think you need to call this with keyword-arg syntax.

File: bigquery/google/cloud/bigquery/client.py (hunk 1 above)
Comment (alixhami) [reply bigquery/google/cloud/bigquery/client.py]: Fixed with c0a7120

File: bigquery/google/cloud/bigquery/client.py (hunk 2 above)
Comment (alixhami) [reply bigquery/google/cloud/bigquery/client.py]: Fixed with c0a7120

**Pull request merged**